libavfilter/vf_scale_qsv.c

```diff
@@ -558,67 +266,20 @@ static int qsvscale_config_props(AVFilterLink *outlink)
 
 static int qsvscale_filter_frame(AVFilterLink *link, AVFrame *in)
 {
-    AVFilterContext             *ctx = link->dst;
-    QSVScaleContext               *s = ctx->priv;
-    AVFilterLink            *outlink = ctx->outputs[0];
-
-    mfxSyncPoint sync = NULL;
-    mfxStatus err;
-
-    AVFrame *out = NULL;
-    int ret = 0;
-
-    out = ff_get_video_buffer(outlink, outlink->w, outlink->h);
-    if (!out) {
-        ret = AVERROR(ENOMEM);
-        goto fail;
-    }
-
-    do {
-        err = MFXVideoVPP_RunFrameVPPAsync(s->session,
-                                           (mfxFrameSurface1*)in->data[3],
-                                           (mfxFrameSurface1*)out->data[3],
-                                           NULL, &sync);
-        if (err == MFX_WRN_DEVICE_BUSY)
-            av_usleep(1);
-    } while (err == MFX_WRN_DEVICE_BUSY);
-
-    if (err < 0) {
-        ret = ff_qsvvpp_print_error(ctx, err, "Error during scaling");
-        goto fail;
-    }
-
-    if (!sync) {
-        av_log(ctx, AV_LOG_ERROR, "No sync during scaling\n");
-        ret = AVERROR_UNKNOWN;
-        goto fail;
-    }
+    int               ret = 0;
+    AVFilterContext  *ctx = link->dst;
+    QSVVPPContext    *qsv = ctx->priv;
+    AVFilterLink *outlink = ctx->outputs[0];
 
-    do {
-        err = MFXVideoCORE_SyncOperation(s->session, sync, 1000);
-    } while (err == MFX_WRN_IN_EXECUTION);
-    if (err < 0) {
-        ret = ff_qsvvpp_print_error(ctx, err, "Error synchronizing the operation");
-        goto fail;
+    if (qsv->session) {
+        ret = ff_qsvvpp_filter_frame(qsv, link, in);
+        av_frame_free(&in);
+    } else {
+        if (in->pts != AV_NOPTS_VALUE)
+            in->pts = av_rescale_q(in->pts, link->time_base, outlink->time_base);
+        ret = ff_filter_frame(outlink, in);
```

Comment:
why a scale filter needs pts conversion?

Reply:
You are right, pts conversion is not needed here.  The intention of adding the code here is to make it similar to the function in vf_vpp_qsv.c, and then re-using all functions for scale_qsv filter.